hudi-utilities/src/main/java/org/apache/hudi/utilities/checkpointing/InitialCheckPointProvider.java

```diff
@@ -18,14 +18,38 @@
 
 package org.apache.hudi.utilities.checkpointing;
 
+import org.apache.hudi.common.config.TypedProperties;
 import org.apache.hudi.exception.HoodieException;
 
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
 /**
  * Provide the initial checkpoint for delta streamer.
  */
-public interface InitialCheckPointProvider {
+public abstract class InitialCheckPointProvider {
+  protected transient Path path;
+  protected transient FileSystem fs;
+  protected transient TypedProperties props;
+
+  static class Config {
+    private static String CHECKPOINT_PROVIDER_PATH_PROP = "hoodie.deltastreamer.checkpoint.provider.path";
+  }
+
+  public InitialCheckPointProvider(TypedProperties props) {
+    this.props = props;
+    this.path = new Path(props.getString(Config.CHECKPOINT_PROVIDER_PATH_PROP));
+  }
+
+  /**
+   * Initialize the class with the current filesystem.
+   *
+   * @param fileSystem
+   */
+  public abstract void init(FileSystem fileSystem);
```

Comment:
would `Configuration` be a better choice here? is more general and the user can construct the filesystem from that? It will also carry other configurations picked up sparkContext say aws creds etc

Reply:
Good point. now it's not `hiveConf` any more :)